onnxruntime/python/tools/transformers/models/sam2/image_encoder.py

```diff
@@ -113,17 +114,65 @@ def export_image_encoder_onnx(
         if not verbose:
             warnings.filterwarnings("ignore", category=torch.jit.TracerWarning)
             warnings.filterwarnings("ignore", category=UserWarning)
-        torch.onnx.export(
-            sam2_encoder,
-            image,
-            onnx_model_path,
-            export_params=True,
-            opset_version=17,
-            do_constant_folding=True,
-            input_names=["image"],
-            output_names=["image_features_0", "image_features_1", "image_embeddings"],
-            dynamic_axes=dynamic_axes,
-        )
+
+        if not dynamo:
+            torch.onnx.export(
+                sam2_encoder,
+                image,
+                onnx_model_path,
+                export_params=True,
+                opset_version=17,
+                do_constant_folding=True,
+                input_names=["image"],
+                output_names=["image_features_0", "image_features_1", "image_embeddings"],
+                dynamic_axes=dynamic_axes,
+            )
+        else:
+            torch._dynamo.config.capture_scalar_outputs = True
+            ep = torch.export.export(
+                sam2_encoder,
+                args=(image,),
+                strict=False,
+                dynamic_shapes=[
+                    {0: torch.export.Dim.AUTO},
+                ],
+            )
+
+            onnx_program = torch.onnx.export(
+                ep,
+                (),
+                opset_version=17,
+                input_names=["image"],
+                output_names=["image_features_0", "image_features_1", "image_embeddings"],
+                dynamo=True,
+            )
+            onnx_program.optimize()
+            onnx_program.save(onnx_model_path + ".dynamo.onnx", external_data=False)
+            import onnx
+
+            from onnxruntime.transformers.dynamo_onnx_helper import DynamoOnnxHelper
+
+            onnx_model = onnx.load_model(onnx_model_path + ".dynamo.onnx", load_external_data=True)
```

Comment:
why we need save model and then load model? can we consume `onnx_program` directly?

Reply:
https://pytorch.org/docs/stable/onnx_dynamo.html#torch.onnx.ONNXProgram does not expose an API to get the onnx model object we needed.